### ISSUE TITLE
"--install-types not supported in this mode of running mypy" even with `--non-interactive`

### ISSUE THREAD

**Issue by y120** [python/mypy#11103]
**Bug Report**

When running mypy with redirected `stdout` / `stderr` (for example, via `mypy.api.run`), `--install-types` fails with:

    error: --install-types not supported in this mode of running mypy

Per [the comment](https://github.com/python/mypy/blob/3ef8040a4c9b9ad05368f593749125b4c7f8067d/mypy/main.py#L73), this is because `--install-types` performs user input. However, when `--non-interactive` is specified, no user input is necessary, and this error should not be raised.

**To Reproduce**

1. Create a file `src/foo.py` like:
```python
# src/foo.py
from mypy.api import run

_, _, exit_code = run(["--install-types", "--non-interactive", "src"])
assert exit_code == 0
```
2. Run `python src/foo.py`

**Expected Behavior**

The program exits with exit code 0.

**Actual Behavior**

```
Traceback (most recent call last):
  File "src/foo.py", line 4, in <module>
    assert exit_code == 0
AssertionError
```

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: `--install-types --non-interactive src` (invoked via api)
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: Python 3.7.5
- Operating system and version: Ubuntu 18.04.5 LTS (Bionic Beaver)
